torch/optim/adamax.py

```diff
@@ -268,13 +275,16 @@ def _single_tensor_adamax(
         # Update biased first moment estimate.
         exp_avg.lerp_(grad, 1 - beta1)
         # Update the exponentially weighted infinity norm.
-        norm_buf = torch.cat(
-            [exp_inf.mul_(beta2).unsqueeze(0), grad.abs().add_(eps).unsqueeze_(0)], 0
-        )
-
         if not differentiable:
-            torch.amax(norm_buf, 0, keepdim=False, out=exp_inf)
+            torch.maximum(
+                exp_inf.mul_(beta2),
+                grad.abs_().add_(eps),
+                out=exp_inf,
+            )
         else:
+            norm_buf = torch.cat(
```

Comment:
maximum didn't work for differentiable case. so fallback to prev

Reply:
Related issue: https://github.com/pytorch/pytorch/issues/54216